/*
 * Copyright 2021 Michael P. Nitowski
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at

       http://www.apache.org/licenses/LICENSE-2.0

 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

syntax = "proto3";

// TODO Configure for gRPC gateway
// import "google/api/annotations.proto";

package circlet.v1;

// option go_package = "github.com/tourneycloud/circlet/gen/go/circlet/v1";

message Redirect {
    string from_location = 1;
    string to_location = 2;
}

message CreateRedirectRequest {
    Redirect redirect = 1;
}

service RedirectService {
    rpc CreateShortenedRedirect(CreateRedirectRequest) returns (Redirect)  {
        // option (google.api.http) = {
        //     post: "/v1/redirects:shorten"
        //     body: "redirect"
        // };
    }
}
